### PR TITLE
Prevent Chartbeat Re-initialisation between article navigations

### LINKED ISSUE #3621: Move chartbeat out of unmount and refactor to avoid unwanted reloading
**Is your feature request related to a problem? Please describe.**
Our current chartbeat code is run inside the article main. This means it gets unmounted when you navigate to a new article (due to the loading screen).

Currently the unmounting logic causes issues as it reinitialises charbeat by reloading the script again.

It also doesnt call `window.pSUPERFLY.virtualPage` in the right place, calling it on unmount instead of when a new article is loaded.

**Describe the solution you'd like**
Move the chartbeat logic into pageWrapper or contexts, (or anouther new layer above `withLoading` to avoid being unmounted. 

**Possible sollution:**
Move it into `UserContext` (or anouther context), creating a `useChartbeat` function which takes `config` as an argument. When `useChartbeat` is first called it renders the scripts using helmet (and never unmounts them). 

If the `config` argument changes (data changes), the `useChartbeat` function will call `window.pSUPERFLY.virtualPage` for you.

We need to be very careful to use the `useEffect` hook properly. The `config` object should only change when `data` changes, not on renders. If this is done wrong, even if `config` is regenerated the same across the renders, `useEffect` will see it as a new object, calling `window.pSUPERFLY.virtualPage` too many times

**Testing notes**
This issue will require a complete testing of chartbeat functionality on articles pages when this refactor is done.

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [x] This feature is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by andrewscfc**
Resolves #3621 

**Overall change:** Adjusted the canonical implementation of Chartbeat Analytics to not reinitialise as the user navigates between articles. This should prevent any duplicate ping events being despatched as the user moves between articles.

**Code changes:**

- `CanonicalChartbeatBeacon` changed to record the chartbeat config on first load as state: https://github.com/bbc/simorgh/pull/4793/files#diff-f84aef1b2ffa544a85d695f08099cfeaR8
- The config recorded in state is used to [render the inline config](https://github.com/bbc/simorgh/pull/4793/files#diff-f84aef1b2ffa544a85d695f08099cfeaR26). This means the chartbeat helment script will not render again after this state has been populated (docs on how to render chartbeat on first visit: http://support.chartbeat.com/docs/#intro)
- The`CanonicalChartbeatBeacon` has [an effect](https://github.com/bbc/simorgh/pull/4793/files#diff-f84aef1b2ffa544a85d695f08099cfeaR12) that is used primarily to dispatch virtual page events, on mount the effect records that the component has mounted in a ref but does not dispatch a virtual page event; inline config above takes care of the ping for the first article.
- When `ArticleMain` is updated with new data, [`ChartbeatAnalytics`](https://github.com/bbc/simorgh/blob/latest/src/app/containers/ArticleMain/index.jsx#L49) is not re-rendered because it continues to use the intial config for it's DOM render
- `CanonicalChartbeatBeacon`'s effect despatches [a virtual page event](https://github.com/bbc/simorgh/pull/4793/files#diff-f84aef1b2ffa544a85d695f08099cfeaR15) , referring to the ref indicating the component has previously mounted; this will be the case for all subsequent journeys between articles.
- Added [test](https://github.com/bbc/simorgh/pull/4793/files#diff-e18a86695298a98884a20b63c2e2e6ddR58) to check the inline script for chartbeat is unchanged between renders, ensuring chartbeat is not re-initialised between renders.

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [x] This PR requires manual testing
Original issue requested this.

### CODE REVIEW

**Comment by andrewscfc** [comment]
As previously, I think the code climate issue is not helpful here as the config is repeated to aid readability. (Happy to discuss :) )

**Review by rhenshaw56 — APPROVED** [review]
Tested this locally, works fine.

File: src/app/containers/ChartbeatAnalytics/canonical/index.jsx [hunk 1]
```diff
@@ -1,26 +1,29 @@
...
-  const chartbeatConfigRef = useRef(chartbeatConfig);
+  const [firstLoadConfig] = useState(chartbeatConfig);
+
+  const hasMounted = useRef(false);
 
   useEffect(() => {
-    if (chartbeatConfigRef.current !== chartbeatConfig) {
+    if (hasMounted.current) {
       // eslint-disable-next-line no-unused-expressions
       window.pSUPERFLY && window.pSUPERFLY.virtualPage(chartbeatConfig);
```
Comment (jroebu14) [inline src/app/containers/ChartbeatAnalytics/canonical/index.jsx]: can we move this `window.pSUPERFLY` into the conditional and then remove the eslint disable line?

**Review by jamesdonoh — APPROVED** [review]
Looks good from my pov, thanks for detailed summary of code changes and links. My only question is whether we think this will work equally well for (currently hypothetical) links across different page types, eg front page to article.

File: src/app/containers/ChartbeatAnalytics/canonical/index.jsx (hunk 1 above)
Comment (andrewscfc) [reply src/app/containers/ChartbeatAnalytics/canonical/index.jsx]: Should now be resolved @jroebu14

**Comment by andrewscfc** [comment]
> Looks good from my pov, thanks for detailed summary of code changes and links. My only question is whether we think this will work equally well for (currently hypothetical) links across different page types, eg front page to article.

@jamesdonoh , I think it should 'just work', it's already [included in the frontpage container](https://github.com/bbc/simorgh/blob/latest/src/app/containers/FrontPageMain/index.jsx#L39) and it will freeze the script tag with the initial config as on an article page, then use the virtual page api. Unless there is another aspect of this you are thinking about?

**Comment by PriyaKR** [comment]
Complete chartbeat testing is required as mentioned on the jira ticket https://jira.dev.bbc.co.uk/browse/NEWSART-605

**Comment by PriyaKR** [comment]
Looks good to me on article pages and home page.

The parameters are being logged as mentioned on the above jira ticket for every ping when on a page.Also the chartbeat virtual update happens once when on a page looking at the parameter 't'.

**Pull request merged**